src/common/Debug.ts

```diff
@@ -14,70 +14,17 @@
  * limitations under the License.
  */
 
-import { isNode } from '../environment.js';
+import Config from './Config.js';
 
-/**
- * A debug function that can be used in any environment.
- *
- * @remarks
- *
- * If used in Node, it falls back to the
- * {@link https://www.npmjs.com/package/debug | debug module}. In the browser it
- * uses `console.log`.
- *
- * @param prefix - this will be prefixed to each log.
- * @returns a function that can be called to log to that debug channel.
- *
- * In Node, use the `DEBUG` environment variable to control logging:
- *
- * ```
- * DEBUG=* // logs all channels
- * DEBUG=foo // logs the `foo` channel
- * DEBUG=foo* // logs any channels starting with `foo`
- * ```
- *
- * In the browser, set `window.__PUPPETEER_DEBUG` to a string:
- *
- * ```
- * window.__PUPPETEER_DEBUG='*'; // logs all channels
- * window.__PUPPETEER_DEBUG='foo'; // logs the `foo` channel
- * window.__PUPPETEER_DEBUG='foo*'; // logs any channels starting with `foo`
- * ```
- *
- * @example
- * ```
- * const log = debug('Page');
- *
- * log('new page created')
- * // logs "Page: new page created"
- * ```
- */
-export const debug = (prefix: string): ((...args: unknown[]) => void) => {
-  if (isNode) {
-    // eslint-disable-next-line @typescript-eslint/no-var-requires
-    return require('debug')(prefix);
-  }
-
-  return (...logArgs: unknown[]): void => {
-    const debugLevel = globalThis.__PUPPETEER_DEBUG as string;
-    if (!debugLevel) return;
+const _debugs = new Map();
```

Comment:
I think this could do with some comments around it - took me a while to realise you're storing debugs for a given prefix.

Reply:
you're right, I'll give it some comments.
Because I can't use `Config.debug` in the top level of a module, it's not initialized at this time, so I define this `debug` method to define real debug logger lazily.